File: demo-matlab/demo_seuif97.m
% The demo using libseuif97 in MATLAB for Windows 64
%
%  1 Put the following files in the path `\extern\seuif97` of the installed MATLAB
%    
%    libseuif97.dll
%    seuif97.h
%    seuif97.m
%
%  2 Add the path to the Search Path of MATLAB
%
%  License: this code is in the public domain
%
%  Author:   Cheng Maohua
%  Email:    cmh@seu.edu.cn
%
% Last modified: 2019.01.08
%

myfuns = seuif97;
p=18.0;
t=535;
h=myfuns.pt(p,t,4);
s=myfuns.pt(p,t,5);
v=myfuns.pt(p,t,3);
fprintf('(p,t),h,s,v: %.2f,%.2f,%.2f,%.4f,%.4f\n',p,t,h,s,v);

File: demo/demo-java/demoseuif97.java
/*
 
 SEUIF97 Library:
 
  1 Windows: C:/Windows/system/libseuif97.dll
  2 Linux:  /usr/lib/libseuif97.so
  3 The Module: ./seuif97.java

Java Native Access (JNA) 
  ./jna.jar

Build:

 javac -cp jna.jar seuif97.java demoseuif97.java
 java -cp .;jna.jar demoseuif97

License: this code is in the public domain

Author:   Cheng Maohua
Email:    cmh@seu.edu.cn

Last modified: 2019.01.07

*/
public class demoseuif97 {

    public static void main(String[] args){
        double p=16.0;
        double t=540.0;
        double h;
        h=seuif97.INSTANCE.seupt(p,t,4);
        System.out.printf("(p,t)->h: (%.1f %.1f) h: %.2f\n",p,t,h);
        t=seuif97.INSTANCE.seuph(p,h,1);
        System.out.printf("(p,h)->t: (%.1f %.2f) t: %.2f ",p,h,t);
   }
}